contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,277 @@
+ * Can currently only creates WMS layers (internal or external) and WMTS layers.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {ol.layer.Layer} The OpenLayers layer or group for the node.
+ * @export
+ */
+gmf.LayertreeController.prototype.getLayer = function(node) {
+  var wmsUrl = this.wmsUrl_;
+  var layerHelper = this.layerHelper_;
+
+  var makeLayers = function(node) {
+    var i, children = node.children;
+    var layers = new ol.Collection();
+    if (goog.isDef(children)) {
+      for (i = 0; i < children.length; i++) {
+        layers.push(makeLayers(children[i]));
+      }
+      return layerHelper.createBasicGroup(layers);
+    }
+
+    if (node.type === 'WMTS') {
+      var layerWithoutSource = new ol.layer.Tile();
+      var promise = layerHelper.createWMTSLayerFromCapabilitites(node.url,
+          node.name);
+      promise.then(function(layer) {
+        if (goog.isDef(layer)) {
+          layerWithoutSource.setSource(layer.getSource());
+        }
+      });
+      return layerWithoutSource;
+
+    } else if (node.type === 'external WMS') {
+      return layerHelper.createBasicWMSLayer(node.url, node.name);
+
+    } else {
+      return layerHelper.createBasicWMSLayer(wmsUrl, node.name);
+    }
+  };
+
+  return makeLayers(node);
+};
+
+
+/**
+ * Return a 'light' class if the current resolution of the map is out of
+ * the  min/max resolution in the node.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {?string} 'light' or null.
+ * @export
+ */
+gmf.LayertreeController.prototype.getResolutionStyle = function(node) {
+  var style;
+  var resolution = this.map.getView().getResolution();
+  var maxExtent = node['maxResolutionHint'];
+  var minExtent = node['minResolutionHint'];
+  if (goog.isDef(minExtent) && resolution < minExtent ||
+      goog.isDef(maxExtent) && resolution > maxExtent) {
+    style = 'light';
+  }
+  return style || null;
+};
+
+
+/**
+ * Toggle activation of a node by adding or removing relative(s) layer(s) on
+ * the map.
+ * @param {ngeo.LayertreeController} treeCtrl Layer tree controller, from the
+ *     current node.
+ * @export
+ */
+gmf.LayertreeController.prototype.toggleActive = function(treeCtrl) {
+  var layer = treeCtrl.scope['layer'];
+  // Check if the current node state is 'activated'.
+  var add = (this.getSetActive(treeCtrl) === 'on') ? false : true;
+  // Add/remove layers
+  var layers = this.layerHelper_.getFlatLayers(layer);
+  this.layerHelper_.moveInOutLayers(layers, add);
+};
+
+
+/**
+ * Return a class name that match with the current node activation state.
+ * @param {ngeo.LayertreeController} treeCtrl Layer tree controller, from the
+ *     current node.
+ * @return {string} The name of de class to apply (on/off/indeterminate).
+ * @export
+ */
+gmf.LayertreeController.prototype.getSetActive = function(treeCtrl) {
+  var style;
+  var node = treeCtrl.node;
+  var layer = treeCtrl.scope['layer'];
```

Comment:
I'm a bit sceptical about this. How will we get the layer from outside the `layertree` directive context ?
I wonder if we wouldn't need to attach the layer to the node, or to have a service that retrieve the layer from the node like https://github.com/Geoportail-Luxembourg/geoportailv3/blob/9e5d18fad4e14befd5795fd1ca10388583f420f6/geoportailv3/static/js/layerfactoryservice.js#L172.

But we can see if we need that later, just for memo.